plugins/kotlin-as-java/src/main/kotlin/converters/KotlinCompanion.kt

```diff
@@ -0,0 +1,78 @@
+package org.jetbrains.dokka.kotlinAsJava.converters
+
+import org.jetbrains.dokka.links.Callable
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.model.properties.PropertyContainer
+
+private const val DEFAULT_COMPANION_NAME = "Companion"
+
+internal class KotlinCompanion(private val companion: DObject?) {
+
+    fun staticFunctions(): List<DFunction> {
+        if(companion == null) return emptyList()
+        return companion.functions.filter { it.isJvmStatic }
+    }
+
+    /**
+     * @return properties that will be visible as static for java.
+     * See [Static fields](https://kotlinlang.org/docs/java-to-kotlin-interop.html#static-fields)
+     */
+    fun staticProperties(): List<DProperty> {
+        if(companion == null) return emptyList()
+        return companion.properties.filter { it.isJvmField || it.isConst || it.isLateInit}
+    }
+
+    fun companionInstancePropertyInJava(): List<DProperty>{
+        if(companion == null) return emptyList()
+        if(companion.hasNothingToRender()) return emptyList() // do not show if companion not rendered
+
+        with(companion){
```

Comment:
If `companionInstancePropertyInJava` will become an extension one to `DObject`, then there will be no need in this `with`